components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/utils/PolicyExportManager.java

```diff
@@ -0,0 +1,242 @@
+    private static final String CUSTOM = "custom_";
+    private static final String ZIP = ".zip";
+    private static final String EXPORT_POLICIES = "ExportPolicies";
+    private APIMgtAdminService apiMgtAdminService;
+
+    public PolicyExportManager(APIMgtAdminService adminService) {
+        apiMgtAdminService = adminService;
+    }
+
+    public String createArchiveFromExecutionPlans(String exportedPoliciesDirName, String archiveDir, String archiveName)
+            throws APIManagementException {
+
+        try {
+            //retrieve all policies under each policy level
+            List<APIPolicy> apiPolicies = apiMgtAdminService.getApiPolicies();
+            List<ApplicationPolicy> applicationPolicies = apiMgtAdminService.getApplicationPolicies();
+            List<SubscriptionPolicy> subscriptionPolicies = apiMgtAdminService.getSubscriptionPolicies();
+            List<CustomPolicy> customPolicies = apiMgtAdminService.getCustomRules();
+            //write all execution Plans/Siddhi Apps to exportPoliciesDirName directory
+            String rootFolder = exportedPoliciesDirName;
+            exportedPoliciesDirName = exportedPoliciesDirName + File.separator + EXPORT_POLICIES;
+            APIFileUtils.createDirectory(exportedPoliciesDirName);
+            if (!apiPolicies.isEmpty()) {
+                for (Map<String, String> map : getApiPolicySiddhiApps(apiPolicies)) {
+                    prepareFile(exportedPoliciesDirName, map);
+                }
+            }
+            if (!applicationPolicies.isEmpty()) {
+                prepareFile(exportedPoliciesDirName, getAppPolicySiddhiApps(applicationPolicies));
+            }
+            if (!subscriptionPolicies.isEmpty()) {
+                prepareFile(exportedPoliciesDirName, getSubscriptionPolicySiddhiApps(subscriptionPolicies));
+            }
+            if (!customPolicies.isEmpty()) {
+                prepareFile(exportedPoliciesDirName, getCustomPolicySiddhiApps(customPolicies));
+            }
+            //create archive and get the archive location
+            String zippedFilePath = createArchiveFromPolicies(rootFolder, archiveDir, archiveName);
+            APIFileUtils.deleteDirectory(rootFolder);
+            return zippedFilePath;
+        } catch (APIManagementException e) {
+            String errorMessage = "Error while exporting policies";
+            log.error(errorMessage, e);
+            throw new APIManagementException(errorMessage, e);
+        }
+    }
+
+    /**
+     * Create Archive from policies.
+     *
+     * @param sourceDir   Source directory containing execution plans
+     * @param archiveDir  Archive containing directory
+     * @param archiveName name of the archive
+     * @return String object of the file path
+     * @throws APIMgtDAOException if creating archive failed
+     */
+    private String createArchiveFromPolicies(String sourceDir, String archiveDir, String archiveName)
+            throws APIMgtDAOException {
```

Comment:
We should not throw APIMgtDAOException from here. wrap it up with an appropriate specific Exception.